src/socket/unix_socket.cr

```diff
@@ -13,31 +13,38 @@
 # ```
 class UNIXSocket < Socket
   getter path : String?
+  getter? abstract : Bool
 
-  # Connects a named UNIX socket, bound to a filesystem pathname.
-  def initialize(@path : String, type : Type = Type::STREAM)
+  # Connects a named UNIX socket.
+  #
+  # A non-abstract socket is bound to a filesystem pathname.
+  def initialize(@path : String, type : Type = Type::STREAM, @abstract = false)
     super(Family::UNIX, type, Protocol::IP)
 
-    connect(UNIXAddress.new(path)) do |error|
+    connect(UNIXAddress.new(path, @abstract)) do |error|
       close
       raise error
     end
   end
 
-  protected def initialize(family : Family, type : Type)
+  # FIXME: NEED REVIEW: what to do with abstract here ?
+  # it is called from UNIXServer#new, where @abstract is already initialized
+  # but if I remove the abstract argument here, I get the error saying that
+  # @abstract is not initialized in all initialize... (See issue #4055)
+  protected def initialize(family : Family, type : Type, @abstract = false)
```

Comment:
I don't see what's to fix. UNIXServer shouldn't set `@abstract` but pass `abstract` to the `super` call, so it will be set, here.